pkg/deployer/utils/utils.go

```diff
@@ -0,0 +1,60 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	lsv1alpha1helper "github.com/gardener/landscaper/apis/core/v1alpha1/helper"
+)
+
+// HandleAnnotationsAndGeneration is meant to be called at the beginning of a deployer's reconcile loop.
+// If a reconcile is needed due to the reconcile annotation or a change in the generation, it will set the phase to Init and remove the reconcile annotation.
+// It will also remove the timeout annotation if it is set.
+// Returns:
+//   - the modified deployitem
+//   - an error, if updating the deployitem failed, nil otherwise
+func HandleAnnotationsAndGeneration(ctx context.Context, c client.Client, log logr.Logger, di *lsv1alpha1.DeployItem) error {
+	changedMeta := false
+	changedStatus := false
+	hasReconcileAnnotation := lsv1alpha1helper.HasOperation(di.ObjectMeta, lsv1alpha1.ReconcileOperation)
+	old := di.DeepCopy()
+	if hasReconcileAnnotation || di.Status.ObservedGeneration != di.Generation {
+		// reconcile necessary
+		log.V(5).Info("reconcile required, setting observed generation and phase", "type", "DeployItem", "name", di.Name, "namespace", di.Namespace, "reconcileAnnotation", hasReconcileAnnotation, "observedGeneration", di.Status.ObservedGeneration, "generation", di.Generation)
```

Comment:
the name and namesapce is duplicated. I suggest to declare a specific logger at the beginning:
```
logger := log.WithValues("name", di.name, "namesapce", di.namespace)
logger.V(5).Info(...)
```